modules/navier_stokes/src/functormaterials/ExponentialFrictionMaterial.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExponentialFrictionMaterial.h"
+#include "MooseMesh.h"
+#include "NS.h"
+
+registerMooseObject("NavierStokesApp", ExponentialFrictionMaterial);
+
+InputParameters
+ExponentialFrictionMaterial::validParams()
+{
+  InputParameters params = FunctorMaterial::validParams();
+  params.addClassDescription("Computes a Reynolds number-exponential friction factor.");
+  params.addRequiredParam<MooseFunctorName>(NS::Reynolds, "The Reynolds number.");
+  params.addParam<MooseFunctorName>(NS::speed, 1.0, "The velocity magnitude of the fluid.");
```

Comment:
What's the reason for defaulting this?

Reply:
Seems like you could remove the default argument and then `paramError` if it's not provided and `_include_velocity_factor`

Reply:
Great point!